django/http/request.py

```diff
@@ -25,7 +26,9 @@
 from .multipartparser import parse_header
 
 RAISE_ERROR = object()
-host_validation_re = _lazy_re_compile(r"^([a-z0-9.-]+|\[[a-f0-9]*:[a-f0-9\.:]+\])(:\d+)?$")
+host_validation_re = _lazy_re_compile(r"^([a-z0-9.-]+|\[[a-f0-9]*:[a-f0-9\.:]+\])(?::([0-9]+))?$")
+
+ParsedHostHeader = namedtuple('ParsedHostHeader', ['domain', 'port', 'combined'])
```

Comment:
`namedtuple`s are weird (indexing syntax) and slow to create (it uses `eval` internally). Can you make a vanilla class, or use dataclasses (not sure if django 3.2 will use python 3.7+ though)?